services/rating/src/main/java/com/eshop/rating/application/querybus/FirstGenericArgOf.java

```diff
@@ -0,0 +1,36 @@
+package com.eshop.rating.application.querybus;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public class FirstGenericArgOf {
+  private final Class<?> aClass;
+
+  FirstGenericArgOf(Class<?> aClass) {
+    this.aClass = aClass;
+  }
+
+  boolean isAssignableFrom(Class<?> otherClass) {
+    Type[] interfaces = aClass.getGenericInterfaces();
+    Type genericSuperclass = aClass.getGenericSuperclass();
+
+    ParameterizedType type;
+    if (interfaces.length > 0) {
+      type = (ParameterizedType) interfaces[0];
+    } else {
+      type = (ParameterizedType) genericSuperclass;
+    }
```

Comment:
Isn't better to have it like this:
ParameterizedType type = interfaces.length > 0 ? (ParameterizedType) interfaces[0] : (ParameterizedType) genericSuperclass;